napalm_ros/ros.py

```
@@ -0,0 +1,800 @@
+import napalm_base.utils.string_parsers
+from yandc import mikrotik
+
+
+class ROSDriver(NetworkDriver):
+    def __init__(self, hostname, username, password, timeout=60, optional_args=None):
+        self.device = None
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.timeout = timeout
+
+        self.candidate_config = []
+        self.merge_config = False
+
+        if isinstance(optional_args, dict):
+            self.snmp_community = optional_args.get('snmp_community', None)
+            self.snmp_port = optional_args.get('snmp_port', 161)
+
+    def cli(self, *commands):
+        cli_output = {}
+        for command in commands:
+            kwargs = {}
+            if command[0] == '/':
+                    run_command = command
+            else:
+                find_index = command.find('=/')
+                if find_index == -1:
+                    raise CommandErrorException('Invalid command - [{}]'.format(command))
+                option, run_command = command.split('=/', 1)
+                if option == 'cache':
+                    kwargs['use_cache'] = True
+                run_command = '/{}'.format(run_command)
+            try:
+                device_output = self.device.cli_command(run_command, **kwargs)
+            except Exception as e:
+                raise CommandErrorException(e)
+            if len(device_output) == 1:
+                if self.device.is_cli_error(device_output[0]):
+                    raise CommandErrorException(device_output[0])
+            cli_output[unicode(command)] = device_output
+        return cli_output
+
+    def close(self):
+        if hasattr(self, 'device'):
+            self.device.disconnect()
+            del self.device
+
+    def commit_config(self):
+        if self.merge_config:
+            self.device.configure_via_cli(self.candidate_config)
+            self.discard_config()
+        else:
+            raise NotImplementedError
```

it's probably dead code as load_replace should raise the Exception before you can hit this line.